python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -751,6 +751,119 @@ interface {{ ethernet_interface.name }}
+{%             if ethernet_interface.isis_authentication.level_1.mode == "sha" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " key-id " ~ ethernet_interface.isis_authentication.level_1.sha.key_id %}
+{%             elif ethernet_interface.isis_authentication.level_1.mode == "shared-secret" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " profile " ~ ethernet_interface.isis_authentication.level_1.shared_secret.profile ~
+                      " algorithm " ~ ethernet_interface.isis_authentication.level_1.shared_secret.algorithm %}
+{%             endif %}
+{%             if ethernet_interface.isis_authentication.level_1.rx_disabled is arista.avd.defined(true) %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " rx-disabled" %}
+{%             endif %}
+   {{ isis_auth_cli }} level-1
+{%         endif %}
+{%         if ethernet_interface.isis_authentication.level_2.mode is arista.avd.defined
+               and (ethernet_interface.isis_authentication.level_2.mode in ["md5", "text"]
+               or (ethernet_interface.isis_authentication.level_2.mode == "sha" and ethernet_interface.isis_authentication.level_2.sha.key_id is arista.avd.defined)
+               or (ethernet_interface.isis_authentication.level_2.mode == "shared-secret"
+                   and ethernet_interface.isis_authentication.level_2.shared_secret.profile is arista.avd.defined
+                   and ethernet_interface.isis_authentication.level_2.shared_secret.algorithm is arista.avd.defined)) %}
+{%             set isis_auth_cli = "isis authentication mode " ~ ethernet_interface.isis_authentication.level_2.mode %}
+{%             if ethernet_interface.isis_authentication.level_2.mode == "sha" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " key-id " ~ ethernet_interface.isis_authentication.level_2.sha.key_id %}
+{%             elif ethernet_interface.isis_authentication.level_2.mode == "shared-secret" %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " profile " ~ ethernet_interface.isis_authentication.level_2.shared_secret.profile ~
+                      " algorithm " ~ ethernet_interface.isis_authentication.level_2.shared_secret.algorithm %}
+{%             endif %}
+{%             if ethernet_interface.isis_authentication.level_2.rx_disabled is arista.avd.defined(true) %}
+{%                 set isis_auth_cli = isis_auth_cli ~ " rx-disabled" %}
+{%             endif %}
+   {{ isis_auth_cli }} level-2
+{%         endif %}
+{%     endif %}
+{%     if ethernet_interface.isis_authentication is arista.avd.defined %}
+{%         set both_key_ids = [] %}
+{%         if ethernet_interface.isis_authentication.both.key_ids is arista.avd.defined %}
+{%             for auth_key in ethernet_interface.isis_authentication.both.key_ids | arista.avd.natural_sort("id") %}
+{%                 if auth_key.id is arista.avd.defined
```

Comment:
as all of these are required keys, we can remove this check

Reply:
Removed.